frontend/src/lib/overlay/index.ts

```diff
@@ -61,6 +74,7 @@ function buildSharedClickOverlay(): { overlayTarget: OverlayAction; overlayConta
     return {
       destroy(): void {
         closeOverlay();
+        document.removeEventListener('click', closeHandler);
```

Comment:
there's a missmatch between when we add the click handler to the doc and when we remove it. We add the click handler once esseintly as a singleton when `buildSharedClickOverlay` is created. However we remove that handler when the overlay container is destroyed. But it'll never get added back again.